activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -787,8 +787,8 @@ def supports_rename_column?
         def configure_connection
           variables = @config.fetch(:variables, {}).stringify_keys
 
-          # By default, MySQL 'where id is null' selects the last inserted id; Turn this off.
-          variables["sql_auto_is_null"] = 0
+          # Since MySQL 5.5 sql_auto_is_null is off by default, but if it is set for any reason, we need to turn it off.
+          variables["sql_auto_is_null"] = 0 unless query_value("select @@sql_auto_is_null") == 0
```

Comment:
```suggestion
          variables["sql_auto_is_null"] = 0 unless execute("select @@sql_auto_is_null", "SCHEMA") == 0
```

I see that at the bottom of the method a query is run like this, I'm not sure yet why its behaving differently than `query_value` but it appears the test passes with this